examples/gallery/line/linestyles.py

```diff
@@ -20,23 +20,36 @@
 import numpy as np
 import pygmt
 
-# Generate a sample line for plotting
-x = np.linspace(0, 10, 500)
-y = np.sin(x)
+# Generate a two-point line for plotting
+x = np.array([0, 7])
+y = np.array([9, 9])
 
 fig = pygmt.Figure()
-fig.basemap(region=[0, 10, -3, 3], projection="X15c/8c", frame=["xaf", "yaf", "WSrt"])
+fig.basemap(region=[0, 10, 0, 10], projection="X15c/8c", frame='+t"Line Styles"')
 
 # Plot the line using the default line style
 fig.plot(x=x, y=y)
-
-# Plot the lines using different line styles
-fig.plot(x=x, y=y + 1.5, pen="1p,red,-")
-fig.plot(x=x, y=y + 1.0, pen="2p,blue,.")
-fig.plot(x=x, y=y + 0.5, pen="1p,red,-.")
-
-fig.plot(x=x, y=y - 0.5, pen="2p,blue,..-")
-fig.plot(x=x, y=y - 1.0, pen="3p,tomato,--.")
-fig.plot(x=x, y=y - 1.5, pen="3p,tomato,4_2:2p")
+fig.text(x=x[-1], y=y[-1], text="solid (default)", justify="ML", offset="0.2c/0c")
+
+# plot the line using different line styles
+for linestyle in [
+    "1p,red,-",  # dashed line
+    "1p,blue,.",  # dotted line
+    "1p,lightblue,-.",  # dash-dotted line
+    "2p,blue,..-",  # dot-dot-dashed line
+    "2p,tomato,--.",  # dash-dash-dotted line
+    "2p,tomato,4_2:2p",  # A pattern of 4-point-long line segment and 2-point-gap between segment
+]:
+    y -= 1  # Move the current line down
+    fig.plot(x=x, y=y, pen=linestyle)
+    fig.text(x=x[-1], y=y[-1], text=linestyle, justify="ML", offset="0.2c/0c")
```

Comment:
Could we avoid using a for-loop here as there are some things here like `y -= 1` and `x[-1]` which might not be as readable for beginner Python users. It will make the code longer, but I think it's worth showing things more explicitly.

Reply:
I think these are very basic syntax even for Python beginners. Perhaps only changing `y -= 1` to `y = y - 1` is enough?

Reply:
How about using a dictionary and doing a for-loop through different y values? Something like
```python
for y, linestyle in {1: "1p,red,-", ...}.items():
    fig.plot(...)
    fig.text(...)
```
The x values can be hardcoded since they're always the same (0 to 9). Only the y is changing.

Reply:
Looping over a dictionary seems a more advanced technique. What about using `x[1]` and `y[1]` (or `x[0]` and `y[0]` for left-side labels)?

Reply:
On second thought, I think the for-loop is fine as is.

Went down a rabbit hole of doing quoted lines following https://docs.generic-mapping-tools.org/latest/cookbook/contour-annotations.html, but somehow I couldn't escape the colon `:` character on the tomato line:


```python
fig = pygmt.Figure()
fig.basemap(region=[0, 10, 0, 10], projection="X15c/8c", frame='+t"Line Styles"')

# Plot the line using the default line style
fig.plot(x=x, y=y)
fig.text(x=x[-1], y=y[-1], text="solid (default)", justify="ML", offset="0.2c/0c")

## Plot the line using different line styles
# dashed line
fig.plot(x=x, y=y - 1, pen="1p,red,-", style='qn1:+l" 1p,red,- "')
# dotted line
fig.plot(x=x, y=y - 2, pen="1p,blue,.", style='qn1:+l" 1p,blue,. "')

# dash-dotted line
fig.plot(x=x, y=y - 3, pen="1p,lightblue,-.", style='qn1:+l" 1p,lightblue,-. "')
# dot-dot-dashed line
fig.plot(x=x, y=y - 4, pen="2p,blue,..-", style='qn1:+l" 2p,blue,..- "')

# dash-dash-dotted line
fig.plot(x=x, y=y - 5, pen="2p,tomato,--.", style='qn1:+l" 2p,tomato,--. "')
# A pattern of 4-point-long line segment and 2-point-gap between segment
fig.plot(x=x, y=y - 6, pen="2p,tomato,4_2:2p", style=r'qn1:+l" 2p,tomato,4_2\:2p "')
fig.show()
```
![temp](https://user-images.githubusercontent.com/23487320/97079990-b5514c00-1654-11eb-8a35-e5a68f92e41e.png)